frame/utility/src/lib.rs

```diff
@@ -173,6 +175,7 @@ pub mod pallet {
 					// Return the actual used weight + base_weight of this call.
 					return Ok(Some(base_weight + weight).into())
 				}
+				Self::deposit_event(Event::ItemCompleted);
```

Comment:
I don't see how these are helpful; afaict if anything goes wrong the `index` + error in `BatchInterrupted` should give enough info?

Reply:
If everything goes right, then it's otherwise impossible to determine the boundary point between two neighbouring dispatches which each emit the same event many an arbitrary amount of times. This is the case presently with staking payouts.